kouncil-backend/src/test/java/com/consdata/kouncil/serde/JsonSchemaDeserializationServiceTest.java

```diff
@@ -0,0 +1,250 @@
+import com.consdata.kouncil.serde.formatter.schema.MessageFormatter;
+import io.confluent.kafka.schemaregistry.client.SchemaRegistryClient;
+import io.confluent.kafka.schemaregistry.json.JsonSchema;
+import lombok.SneakyThrows;
+import org.apache.kafka.clients.consumer.ConsumerRecord;
+import org.apache.kafka.common.record.TimestampType;
+import org.apache.kafka.common.utils.Bytes;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.EnumMap;
+import java.util.Objects;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.ArgumentMatchers.*;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(SpringExtension.class)
+@SpringBootTest
+public class JsonSchemaDeserializationServiceTest {
+    private static final byte[] JSON_SCHEMA_SIMPLE_MESSAGE_BYTES = new byte[]{0, 0, 0, 0, 0, 123, 34, 99, 111, 110, 116, 101, 110, 116, 34, 58, 34, 76, 111, 114, 101, 109, 32, 99, 111, 110, 115, 101, 99, 116, 101, 116, 117, 114, 34, 44, 34, 115, 111, 109, 101, 78, 117, 109, 98, 101, 114, 34, 58, 53, 53, 57, 57, 51, 51, 44, 34, 114, 101, 99, 101, 105, 118, 101, 100, 68, 97, 116, 101, 34, 58, 34, 118, 101, 110, 105, 97, 109, 32, 118, 111, 108, 117, 112, 116, 97, 116, 101, 34, 125};
+    private static final String LOREM = "lorem";
+    private static final String IPSUM = "ipsum";
+    private static final String CLUSTER_ID = "clusterId";
+    private static JsonSchema JSON_SCHEMA;
+    private static String SIMPLE_MESSAGE_JSON;
+    @MockBean
+    private SchemaAwareClusterService schemaAwareClusterService;
+
+    @MockBean
+    private SchemaRegistryFacade schemaRegistryFacade;
+
+    @MockBean
+    private SchemaRegistryClient schemaRegistryClient;
+
+    @Autowired
+    private DeserializationService deserializationService;
+
+    @BeforeAll
+    public static void beforeAll() throws IOException, URISyntaxException {
+        var jsonSchemaPath = Paths.get(JsonSchemaDeserializationServiceTest.class.getClassLoader()
+                .getResource("SimpleMessage.json").toURI());
+        JSON_SCHEMA = new JsonSchema(Files.readString(jsonSchemaPath));
+
+        SIMPLE_MESSAGE_JSON = Files.readString(
+                Paths.get(Objects.requireNonNull(
+                        JsonSchemaDeserializationServiceTest.class.getClassLoader().getResource("SimpleMessage.json")).toURI()
+                )).trim();
+    }
+
+    @Test
+    public void should_deserialize_without_schema() {
```

Comment:
Tak w zasadzie to po co jest ten test i dwa kolejne? Widzę, że są one też w klaskach do deserializacji dla avro i protobuf. Różnią się czymś między sobą? Może warto by było je gdzieś wydzielić albo zostawić tylko w jednej klasie?

Reply:
też zastanawiałem się żeby zrobić jedną klasę, która miałaby wydzielone wspólne rzeczy i wtedy każda by dziedziczyła od niej, to może tak zrobię

Reply:
Bardziej myślałem żeby te dwa czy trzy testy do osobnej klasy przerzucić

Reply:
done, wydzieliłem te bez scheme'y do dwóch osobnych klasek (ser/des)